src/model_config_utils.cc

```diff
@@ -751,6 +751,8 @@ NormalizeInstanceGroup(
       if (pg.kind() == inference::ModelInstanceGroup::KIND_GPU) {
         // Don't use preferred group with KIND_GPU if there is no GPU.
         if (supported_gpus.empty()) {
+          group->set_kind(inference::ModelInstanceGroup::KIND_AUTO);
+          group->set_count(0);
```

Comment:
Should we rather just move line 748-749 down so we only set kind / count once when the preferred group is valid?

Reply:
Addressed here: https://github.com/triton-inference-server/core/pull/233/commits/749e46b013a563bec9042dff6f28b5013874bad9